src/librustdoc/html/render/mod.rs

```diff
@@ -198,11 +197,11 @@ impl SharedContext<'_> {
 
     /// Based on whether the `collapse-docs` pass was run, return either the `doc_value` or the
     /// `collapsed_doc_value` of the given item.
-    crate fn maybe_collapsed_doc_value<'a>(&self, item: &'a clean::Item) -> Option<Cow<'a, str>> {
+    crate fn maybe_collapsed_doc_value<'a>(&self, item: &'a clean::Item) -> Option<String> {
         if self.collapsed {
-            item.collapsed_doc_value().map(|s| s.into())
+            item.collapsed_doc_value().map(|s| s.to_string())
```

Comment:
This already returns `Option<String>`.
```suggestion
            item.collapsed_doc_value()
```